internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -33,19 +33,21 @@ type ContainerNodeConfig struct {
 // computeCostComponent returns a cost component for Compute instance usage.
 func computeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) ([]*schema.CostComponent, error) {
 	if strings.HasPrefix(strings.ToLower(machineType), "e2-custom") {
-		return nil, errors.New("Infracost currently does not support E2 custom instances")
+		return nil, errors.New("infracost currently does not support E2 custom instances")
 	}
 
 	sustainedUseDiscount := 0.0
 	fixPurchaseOption := ""
 
-	if strings.ToLower(purchaseOption) == "on_demand" {
+	if strings.ToLower(purchaseOption) == "on_demand" && monthlyHours != nil {
```

Comment:
We should probably do something like this instead:
```suggestion
        hours := schema.HourToMonthUnitMultiplier
        if monthlyHours != nil {
                hours = *monthlyHours
        }
        
	if strings.ToLower(purchaseOption) == "on_demand" {
```

And then pass in `hours` to `getSustainedUseDiscount`